api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -631,5 +631,19 @@ public void changePassword(User user, String oldPassword, String newPassword) th
 		OpenmrsUtil.validatePassword(user.getUsername(), newPassword, user.getSystemId());
 		dao.changePassword(user, newPassword);
 	}
+
+	@Override
+	public void changePassword(User user, String newPassword) throws APIException {
+		dao.changePassword(user, newPassword);
+	}
+
+	@Override
+	public void changePasswordUsingSecretAnswer(String secretAnswer, String pw) throws APIException {
+		User user = Context.getAuthenticatedUser();
```

Comment:
The code here should pretty much be the same, only different should be that this checks that the secret answer is correct. You could factor the other code in the other changePassword method into a utility method that they both call after checking that the old or secret password is correct